k

```diff
@@ -26,9 +26,21 @@ if [ ! -z "$KUBECONFIG" ]; then
   TARGET_VERSION=$(cat "$VERSION_CACHE_FILE" 2> /dev/null)
 fi
 
+# Ensure we have at least 1 kubectl version
+KNOWN_VERSION="v1.16.3"
```

Comment:
Should we put all of this inside of a block like:

```
if ! hash kubectl 2>/dev/null; then
  KNOWN_VERSION="v1.16.3"
  KNOWN_TARGET=$KX_PATH/kubectl-$KNOWN_VERSION
    if [ ! -f "$KNOWN_TARGET" ]; then
    curl -L -o $KNOWN_TARGET "https://storage.googleapis.com/kubernetes-release/release/$KNOWN_VERSION/bin/$OS/amd64/kubectl"
    chmod +x $KNOWN_TARGET
 fi
fi
```

tl;dr we don't need to download a known version of kubectl if kubectl is already present

Reply:
I had it written that way, but then thought that people who use this tool are ok not caring where their kubectl comes from. As written, this is a bit shorter and has few `if` statements. Happy to do it either way.

Reply:
I'm more thinking about an unnecessary i/o call. The only reason we download 1.16.3 is "in case the user literally has no kubectl at all", right? If so, should we check first "does the user literally have no kubectl at all?"

Reply:
I'm sold. Just a minute.

Reply:
@jackfrancis  take a look now!